src/ReactLeafletMap/index.js

```diff
@@ -70,86 +87,113 @@ class ReactLeafletMap extends React.PureComponent {
+                change.center = {
+                    lat: viewport.center[0],
+                    lon: viewport.center[1]
+                }
             }
-        });
+
+            if (viewport.hasOwnProperty('zoom') && Number.isFinite(viewport.zoom) && viewport.zoom !== this.state.leafletView.zoom) {
+                change.boxRange = viewUtils.getBoxRangeFromZoomLevel(viewport.zoom, this.state.width, this.state.height);
+            }
+
+            if (!_.isEmpty(change) && this.props.onViewChange && !this.hasResized()) {
+                this.props.onViewChange(change);
+            }
+
+            // TODO for IndexedVectorLayer rerender (see IndexedVectorLayer render method)
+            this.setState({viewport});
+        }
+    }
+
+    hasResized() {
+        const {x,y} = this.leafletMap._size;
+        const widthChange = Math.abs(x - this.state.width) > 1;
+        const heightChange = Math.abs(y - this.state.height) > 1;
+        return widthChange || heightChange;
     }
 
-    onViewportChange(viewport) {
-        const change = {};
-        // viewport.center could be undefined
-        if(viewport && viewport.center) {
-            const center = {
-                lat: viewport.center[0],
-                lon: viewport.center[1]
-            };
-            change['center'] = center;
+    onResize(width, height) {
+        if (this.leafletMap) {
+            this.leafletMap.invalidateSize();
         }
-        
-        // viewport.zoom could be undefined
-        if(viewport && viewport.hasOwnProperty('zoom') && Number.isFinite(viewport.zoom)) {
-            const boxRange = viewUtils.getBoxRangeFromZoomLevel(viewport.zoom);
-            change['boxRange'] = boxRange;
+
+        if (!this.maxZoom && !this.minZoom) {
+            this.setZoomLevelsBounds(width, height);
         }
 
-        // TODO for IndexedVectorLayer rerender (see IndexedVectorLayer render method)
-        let stateUpdate = {viewport};
+        this.setState({
+            width, height, leafletView: viewHelpers.getLeafletViewportFromViewParams(this.props.view, this.state.width, this.state.height)
+        });
 
-        if (this.props.onViewChange) {
-            this.props.onViewChange(change);
-        } else {
-            stateUpdate.view = change;
+        if (this.props.onResize) {
+            this.props.onResize(width, height);
         }
-
-        this.setState(stateUpdate);
     }
 
-    onResize() {
-        this.leafletMap.invalidateSize();
+    render() {
+        return (
+            <>
+                <ReactResizeDetector
+                    handleHeight
+                    handleWidth
+                    onResize={this.onResize}
+                    refreshMode="debounce"
+                    refreshRate={500}
+                />
+                {this.state.width && this.state.height ? this.renderMap() : null}
```

Comment:
Will this work with SSR?

Reply:
No idea. We should test this before production version release.